premerge/advisor/git_utils.py

```diff
@@ -0,0 +1,78 @@
+import sqlite3
+import os
+import subprocess
+import logging
+
+REPOSITORY_URL = "https://github.com/llvm/llvm-project"
+FIRST_COMMIT_SHA = "f8f7f1b67c8ee5d81847955dc36fab86a6d129ad"
+
+
+def _clone_repository_if_not_present(
+    repository_path: str, repository_url=REPOSITORY_URL
+):
+    if not os.path.exists(os.path.join(repository_path, ".git")):
+        logging.info("Cloning git repository.")
+        subprocess.run(
+            ["git", "clone", repository_url, os.path.basename(repository_path)],
+            cwd=os.path.dirname(repository_path),
+            check=True,
+        )
+        logging.info("Finished cloning git repository.")
+
+
+def _get_and_add_commit_index(
+    commit_sha: str,
+    repository_path: str,
+    db_connection: sqlite3.Connection,
+    first_commit_sha,
+) -> int:
+    # Ensure the repository is up to date.
+    subprocess.run(["git", "fetch"], cwd=repository_path, check=True)
+    # Get the highest indexed commit so we can ensure we only add new
+    # commits.
+    latest_commit_info = db_connection.execute(
+        "SELECT * FROM commits ORDER BY commit_index DESC"
+    ).fetchone()
+    commits_to_add = []
+    if latest_commit_info:
+        latest_sha, latest_index = latest_commit_info
+    else:
+        latest_sha = first_commit_sha
+        latest_index = 1
+    log_output = subprocess.run(
+        ["git", "log", "--oneline", "--no-abbrev", f"{latest_sha}..{commit_sha}"],
+        cwd=repository_path,
+        stdout=subprocess.PIPE,
+        check=True,
+    )
+    log_lines = log_output.stdout.decode("utf-8").split("\n")[:-1]
```

Comment:
Just wondering: Why are you omitting the last line ? ([:-1])

Reply:
The last line also has a `\n` at the end of the last string, so the last string in the split is just "" which we don't care about.